### PR TITLE
integration: Cleanup, improvements, and switching to systemd as cgroup manager

### PR DESCRIPTION
**Pull request by fidencio**
This series contains is a mix of few clean-ups in the integration tests, and does the switch to using systemd as the cgroup manager.

Clean-ups:
* Removal of old OpenShift tests;
* Removal of old CRI-O tests;
* Simplification of CRI-O installation scripts;

Fixes: #3410, #3411, #3412, and #3413

### CODE REVIEW

File: integration/kubernetes/kubeadm/config.yaml [hunk 1]
```diff
@@ -21,7 +21,7 @@ scheduler:
 ---
 apiVersion: kubelet.config.k8s.io/v1beta1
 kind: KubeletConfiguration
-cgroupDriver: cgroupfs
```
Comment (egernst) [inline integration/kubernetes/kubeadm/config.yaml:24]: is it feasible to use both cgroup managers in our tests? I'd like to ensure cgroupfs is working well still.

Comment (fidencio) [reply integration/kubernetes/kubeadm/config.yaml:24]: It is.  What about testing containerd with cgroupfs (so, no changes on containerd side) and CRI-O using systemd?

Comment (fidencio) [reply integration/kubernetes/kubeadm/config.yaml:24]: it'd be something as simple as this follow-up patch (which I'll apply after a first run of the CI passes).
```
fidencio@quino ~/go/src/github.com/kata-containers/tests $ git status
En la rama wip/integration-tests-improvements
Cambios no rastreados para el commit:
  (usa "git add <archivo>..." para actualizar lo que será confirmado)
  (usa "git restore <archivo>..." para descartar los cambios en el directorio de trabajo)
	modificado:     integration/kubernetes/init.sh
	modificado:     integration/kubernetes/kubeadm/config.yaml

sin cambios agregados al commit (usa "git add" y/o "git commit -a")
fidencio@quino ~/go/src/github.com/kata-containers/tests $ git diff
diff --git a/integration/kubernetes/init.sh b/integration/kubernetes/init.sh
index 700c7def..ab2d77ff 100755
--- a/integration/kubernetes/init.sh
+++ b/integration/kubernetes/init.sh
@@ -55,9 +55,11 @@ fi
 case "${cri_runtime}" in
 containerd)
        cri_runtime_socket="/run/containerd/containerd.sock"
+       cgroup_driver="cgroupfs"
        ;;
 crio)
        cri_runtime_socket="/var/run/crio/crio.sock"
+       cgroup_driver="systemd"
        ;;
 *)
        echo "Runtime ${cri_runtime} not supported"
@@ -100,6 +102,7 @@ kubeadm_config_file="$(mktemp --tmpdir kubeadm_config.XXXXXX.yaml)"
 
 sed -e "s|CRI_RUNTIME_SOCKET|${cri_runtime_socket}|" "${kubeadm_config_template}" > "${kubeadm_config_file}"
 sed -i "s|KUBERNETES_VERSION|v${kubernetes_version/-*}|" "${kubeadm_config_file}"
+sed -i "s|CGROUP_DRIVER|${cgroup_driver}|" "${kubeadm_config_file}"
 
 trap 'sudo -E sh -c "rm -r "${kubeadm_config_file}""' EXIT
 
diff --git a/integration/kubernetes/kubeadm/config.yaml b/integration/kubernetes/kubeadm/config.yaml
index ee1f0b92..f6f013ec 100644
--- a/integration/kubernetes/kubeadm/config.yaml
+++ b/integration/kubernetes/kubeadm/config.yaml
@@ -21,7 +21,7 @@ scheduler:
 ---
 apiVersion: kubelet.config.k8s.io/v1beta1
 kind: KubeletConfiguration
-cgroupDriver: systemd
+cgroupDriver: CGROUP_DRIVER
 featureGates:
   PodOverhead: true
 systemReserved:
```

**Review by wainersm — COMMENTED** [review]
Apart from those changes, I checked you going to remove the openshift reference on version.yaml (https://github.com/kata-containers/kata-containers/pull/1689).

It is  missing the removal of `.ci/install_openshift.sh` and adjust the `.ci/setup.sh` and `.ci/jenkins_job_build.sh`. Same rationale for the clean up of crio.

**Comment by fidencio** [comment]
About Fedora-32-crio-k8s-e2e-minimal and any other thing using sonobouy:
```
19:29:23 time="2021-04-28T17:29:23Z" level=warning msg="The maximum supported Kubernetes version is 1.18.99, but the server version is v1.21.0. Sonobuoy will continue but unexpected results may occur."
```

**Review by chavafg — APPROVED** [review]

File: Makefile [hunk 2]
```diff
@@ -110,10 +110,6 @@ else
 	bash sanity/check_sanity.sh
 endif
 
-crio:
-	bash .ci/install_bats.sh
-	RUNTIME=${RUNTIME} ./integration/cri-o/cri-o.sh
```
Comment (wainersm) [inline Makefile:115]: The CI job http://jenkins.katacontainers.io/job/kata-containers-crio-PR calls that `integration/cri-o/crio-o.sh` to run the cri-io functional tests, however currently is configured for Kata 1.x. While this change has no impact on that job, I am wondering what is the strategy for when we enable a similar job for 2.x. More precisely, even for a 2.x job we won't call that script but instead run a k8s+crio (running the k8s conformance) job. Did I understand it correctly?

Comment (wainersm) [reply Makefile:115]: oh, also I think we won't need the `TEST_CRIO` variable anymore. So need to adjust `.ci/jenkins_job_build.sh`. Such as change I would defer to after today's release.

Comment (fidencio) [reply Makefile:115]: @wainersm, you got that correctly.  The whole integration between CRI-O and Kata Containers will be done differently for 2.x.
The part on the kata-containers repo that we'll use, is based in the `integration/kubernetes`, and the part that uses `integration/cri-o/cri-o.sh` will be implemented entirely in the `prow`, in the OpenShift CI.

Does this make sense? :-)

**Review by wainersm — COMMENTED** [review]
A few typos on the commit message: s/sonobouy/sonobuoy/

File: .ci/install_crio.sh [hunk 3]
```diff
@@ -126,7 +126,7 @@ make clean
 make BUILDTAGS='seccomp selinux exclude_graphdriver_btrfs exclude_graphdriver_devicemapper libdm_no_deferred_remove'
 make test-binaries
 sudo -E PATH=$PATH sh -c "make install"
-sudo -E PATH=$PATH sh -c 'crio -d "" config > crio.conf'
+sudo -E PATH=$PATH sh -c 'crio -d "" --cgroup-manager "systemd" config > crio.conf'
```
Comment (wainersm) [inline .ci/install_crio.sh]: See previous comment, it seems that here we can use `CONF_OVERRIDES` and just call `make install`.

File: .ci/install_crio.sh [hunk 4]
```diff
@@ -148,6 +126,7 @@ make clean
 make BUILDTAGS='seccomp selinux exclude_graphdriver_btrfs exclude_graphdriver_devicemapper libdm_no_deferred_remove'
 make test-binaries
 sudo -E PATH=$PATH sh -c "make install"
```
[the remark this answers is not in the document]
Comment (fidencio) [reply .ci/install_crio.sh:128]: That's a good point.

The reason we do this is that `make install` calls `make install.config`, and that calls `install.config-nobuild, which is:
```
        install ${SELINUXOPT} -d $(DATAROOTDIR)/oci/hooks.d
        install ${SELINUXOPT} -d $(ETCDIR_CRIO)/crio.conf.d
        install ${SELINUXOPT} -D -m 644 crio.conf $(ETCDIR_CRIO)/crio.conf
        install ${SELINUXOPT} -D -m 644 crio-umount.conf $(OCIUMOUNTINSTALLDIR)/crio-umount.conf
        install ${SELINUXOPT} -D -m 644 crictl.yaml $(CRICTL_CONFIG_DIR)
```

That doesn't allow us to configure the options we want in the configuration file.
The workaround to do that (also used as part of the downstream RPM), is to use `crio config` to generate the configuration file and then `make install.config` to install it.

Does it make sense? Am I fine to keep it as it is? :-)

File: .ci/install_crio.sh (hunk 3 above)
Comment (fidencio) [reply .ci/install_crio.sh]: Check the answer for the previous comment. :-)

File: integration/kubernetes/k8s-sysctls.bats [hunk 5]
```diff
@@ -7,15 +7,23 @@
...
+                skip "test not working on CRI-O, see: ${issue}"
+        fi
+
 	export KUBECONFIG="${KUBECONFIG:-$HOME/.kube/config}"
 	pod_name="sysctl-test"
 	get_pod_config_dir
 }
 
 @test "Setting sysctl" {
+        if [ "$CI_JOB" == "CRIO_K8S" ]; then
```
Comment (wainersm) [inline integration/kubernetes/k8s-sysctls.bats:23]: Skipping on `setup` won't let the test run

File: .ci/install_crio.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (wainersm) [reply .ci/install_crio.sh:128]: I thought that https://github.com/cri-o/cri-o/blob/master/Makefile#L63 would be enough. I am not expert on cri-o so keep it as is :)

**Review by wainersm — APPROVED** [review]
@fidencio left a couple of nits which can be addressed on a separate PR after the release (I think I am repeating myself :).

**Comment by fidencio** [comment]
**jenkins-ci-ARM-ubuntu-18-04** failure is not relaed to this PR:
```
21:31:49 🇩 test-integration
21:31:50 === RUN   TestImageLoad
21:31:50     TestImageLoad: image_load_test.go:39: docker save image into tarball
21:31:50     TestImageLoad: image_load_test.go:41: 
21:31:50         	Error Trace:	image_load_test.go:41
21:31:50         	Error:      	Received unexpected error:
21:31:50         	            	exit status 1
21:31:50         	Test:       	TestImageLoad
21:31:50         	Messages:   	output: "Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?\n"
21:31:50 --- FAIL: TestImageLoad (0.30s)
21:31:50 FAIL
21:31:50 ./hack/test-utils.sh: line 82: 18128 Terminated              keepalive "sudo PATH=${PATH} ${ROOT}/_output/${CONTAINERD_BIN} ${CONTAINERD_FLAGS}" ${RESTART_WAIT_PERIOD} &> ${report_dir}/containerd.log
21:31:50 Makefile:95: recipe for target 'test-integration' failed
21:31:50 make[1]: *** [test-integration] Error 1
21:31:50 make[1]: Leaving directory '/home/jenkins/workspace/kata-containers-2.0-tests-ubuntu-ARM-PR/go/src/github.com/containerd/cri'
21:31:50 ERROR: containerd log :
21:31:50 -------------------------------------
21:31:50 sudo PATH=/home/jenkins/workspace/kata-containers-2.0-tests-ubuntu-ARM-PR/go/bin:/usr/local/go/bin:/usr/sbin:/sbin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/usr/local/sbin:/usr/local/bin /home/jenkins/workspace/kata-containers-2.0-tests-ubuntu-ARM-PR/go/src/github.com/containerd/cri/hack/../_output/containerd-test --log-level=debug --config /tmp/test-cri-containerd.ZrOo/test-containerd-config --address /run/containerd-test/containerd.sock --state /run/containerd-test --root /var/lib/containerd-test
21:31:50 time="2021-04-30T19:31:49.820871437Z" level=warning msg="This customized containerd is only for CI test, DO NOT use it for distribution."
21:31:50 time="2021-04-30T19:31:49.832274882Z" level=info msg="starting containerd" revision= version=f864905c-TEST
21:31:50 time="2021-04-30T19:31:49.883117193Z" level=info msg="loading plugin \"io.containerd.content.v1.content\"..." type=io.containerd.content.v1
21:31:50 time="2021-04-30T19:31:49.883428434Z" level=info msg="loading plugin \"io.containerd.snapshotter.v1.overlayfs\"..." type=io.containerd.snapshotter.v1
21:31:50 time="2021-04-30T19:31:49.883963769Z" level=info msg="loading plugin \"io.containerd.metadata.v1.bolt\"..." type=io.containerd.metadata.v1
21:31:50 time="2021-04-30T19:31:49.884117920Z" level=info msg="metadata content store policy set" policy=shared
21:31:50 time="2021-04-30T19:31:49.895605530Z" level=info msg="loading plugin \"io.containerd.differ.v1.walking\"..." type=io.containerd.differ.v1
21:31:50 time="2021-04-30T19:31:49.895930972Z" level=info msg="loading plugin \"io.containerd.gc.v1.scheduler\"..." type=io.containerd.gc.v1
21:31:50 time="2021-04-30T19:31:49.896294897Z" level=info msg="loading plugin \"io.containerd.runtime.v1.linux\"..." type=io.containerd.runtime.v1
21:31:50 time="2021-04-30T19:31:49.897485497Z" level=info msg="loading plugin \"io.containerd.monitor.v1.cgroups\"..." type=io.containerd.monitor.v1
21:31:50 time="2021-04-30T19:31:49.901314113Z" level=info msg="loading plugin \"io.containerd.runtime.v2.task\"..." type=io.containerd.runtime.v2
21:31:50 time="2021-04-30T19:31:49.902811694Z" level=info msg="loading plugin \"io.containerd.service.v1.containers-service\"..." type=io.containerd.service.v1
21:31:50 time="2021-04-30T19:31:49.903098313Z" level=info msg="loading plugin \"io.containerd.service.v1.content-service\"..." type=io.containerd.service.v1
21:31:50 time="2021-04-30T19:31:49.903309007Z" level=info msg="loading plugin \"io.containerd.service.v1.diff-service\"..." type=io.containerd.service.v1
21:31:50 time="2021-04-30T19:31:49.903525062Z" level=info msg="loading plugin \"io.containerd.service.v1.images-service\"..." type=io.containerd.service.v1
21:31:50 time="2021-04-30T19:31:49.903727295Z" level=info msg="loading plugin \"io.containerd.service.v1.leases-service\"..." type=io.containerd.service.v1
21:31:50 time="2021-04-30T19:31:49.904004094Z" level=info msg="loading plugin \"io.containerd.service.v1.namespaces-service\"..." type=io.containerd.service.v1
21:31:50 time="2021-04-30T19:31:49.904228329Z" level=info msg="loading plugin \"io.containerd.service.v1.snapshots-service\"..." type=io.containerd.service.v1
21:31:50 time="2021-04-30T19:31:49.904431803Z" level=info msg="loading plugin \"io.containerd.service.v1.tasks-service\"..." type=io.containerd.service.v1
21:31:50 time="2021-04-30T19:31:49.904678779Z" level=info msg="loading plugin \"io.containerd.grpc.v1.containers\"..." type=io.containerd.grpc.v1
21:31:50 time="2021-04-30T19:31:49.904903554Z" level=info msg="loading plugin \"io.containerd.grpc.v1.content\"..." type=io.containerd.grpc.v1
21:31:50 time="2021-04-30T19:31:49.905107848Z" level=info msg="loading plugin \"io.containerd.grpc.v1.diff\"..." type=io.containerd.grpc.v1
21:31:50 time="2021-04-30T19:31:49.905305141Z" level=info msg="loading plugin \"io.containerd.grpc.v1.events\"..." type=io.containerd.grpc.v1
21:31:50 time="2021-04-30T19:31:49.905495654Z" level=info msg="loading plugin \"io.containerd.grpc.v1.healthcheck\"..." type=io.containerd.grpc.v1
21:31:50 time="2021-04-30T19:31:49.905696927Z" level=info msg="loading plugin \"io.containerd.grpc.v1.images\"..." type=io.containerd.grpc.v1
21:31:50 time="2021-04-30T19:31:49.905895241Z" level=info msg="loading plugin \"io.containerd.grpc.v1.leases\"..." type=io.containerd.grpc.v1
21:31:50 time="2021-04-30T19:31:49.906187360Z" level=info msg="loading plugin \"io.containerd.grpc.v1.namespaces\"..." type=io.containerd.grpc.v1
21:31:50 time="2021-04-30T19:31:49.906463539Z" level=info msg="loading plugin \"io.containerd.grpc.v1.snapshots\"..." type=io.containerd.grpc.v1
21:31:50 time="2021-04-30T19:31:49.906702435Z" level=info msg="loading plugin \"io.containerd.grpc.v1.tasks\"..." type=io.containerd.grpc.v1
21:31:50 time="2021-04-30T19:31:49.906959612Z" level=info msg="loading plugin \"io.containerd.grpc.v1.version\"..." type=io.containerd.grpc.v1
21:31:50 time="2021-04-30T19:31:49.907154825Z" level=info msg="loading plugin \"io.containerd.grpc.v1.cri\"..." type=io.containerd.grpc.v1
21:31:50 time="2021-04-30T19:31:49.908684508Z" level=info msg="Start cri plugin with config {PluginConfig:{ContainerdConfig:{Snapshotter:overlayfs DefaultRuntimeName:runc DefaultRuntime:{Type: Engine: PodAnnotations:[] Root: Options:<nil> PrivilegedWithoutHostDevices:false} UntrustedWorkloadRuntime:{Type: Engine: PodAnnotations:[] Root: Options:<nil> PrivilegedWithoutHostDevices:false} Runtimes:map[runc:{Type:io.containerd.runtime.v1.linux Engine: PodAnnotations:[] Root: Options:0x400011a450 PrivilegedWithoutHostDevices:false}] NoPivot:false} CniConfig:{NetworkPluginBinDir:/opt/cni/bin NetworkPluginConfDir:/etc/cni/net.d NetworkPluginMaxConfNum:1 NetworkPluginConfTemplate:} Registry:{Mirrors:map[docker.io:{Endpoints:[https://registry-1.docker.io]}] Configs:map[] Auths:map[]} DisableTCPService:true StreamServerAddress:127.0.0.1 StreamServerPort:0 StreamIdleTimeout:4h0m0s EnableSelinux:false SandboxImage:k8s.gcr.io/pause:3.1 StatsCollectPeriod:10 SystemdCgroup:false EnableTLSStreaming:false X509KeyPairStreaming:{TLSCertFile: TLSKeyFile:} MaxContainerLogLineSize:16384 DisableCgroup:false DisableApparmor:false RestrictOOMScoreAdj:false MaxConcurrentDownloads:3 DisableProcMount:false} ContainerdRootDir:/var/lib/containerd-test ContainerdEndpoint:/run/containerd-test/containerd.sock RootDir:/var/lib/containerd-test/io.containerd.grpc.v1.cri StateDir:/run/containerd-test/io.containerd.grpc.v1.cri}"
21:31:50 time="2021-04-30T19:31:49.909397776Z" level=info msg="Connect containerd service"
21:31:50 time="2021-04-30T19:31:49.911378489Z" level=info msg="Get image filesystem path \"/var/lib/containerd-test/io.containerd.snapshotter.v1.overlayfs\""
21:31:50 time="2021-04-30T19:31:49.915561329Z" level=info msg="loading plugin \"io.containerd.grpc.v1.introspection\"..." type=io.containerd.grpc.v1
21:31:50 time="2021-04-30T19:31:49.915751162Z" level=info msg="Start subscribing containerd event"
21:31:50 time="2021-04-30T19:31:49.915929214Z" level=info msg="Start recovering state"
21:31:50 time="2021-04-30T19:31:49.915996378Z" level=info msg=serving... address=/run/containerd-test/containerd.sock.ttrpc
21:31:50 time="2021-04-30T19:31:49.916138228Z" level=info msg="Start event monitor"
21:31:50 time="2021-04-30T19:31:49.916182371Z" level=info msg="Start snapshots syncer"
21:31:50 time="2021-04-30T19:31:49.916208973Z" level=info msg="Start streaming server"
21:31:50 time="2021-04-30T19:31:49.916149409Z" level=info msg=serving... address=/run/containerd-test/containerd.sock
21:31:50 time="2021-04-30T19:31:49.916562816Z" level=info msg="containerd successfully booted in 0.094581s"
21:31:50 time="2021-04-30T19:31:49.999425895Z" level=debug msg="garbage collected" d=2.583213ms
21:31:50 time="2021-04-30T19:31:50.523437752Z" level=debug msg="received signal" signal=terminated
21:31:50 time="2021-04-30T19:31:50.524171161Z" level=info msg="Stop CRI service"
21:31:50 -------------------------------------
21:31:50 Makefile:148: recipe for target 'cri-containerd' failed
21:31:50 make: *** [cri-containerd] Error 2
21:31:50 Failed at 24: sudo -E PATH="$PATH" bash -c "make cri-containerd"
21:31:51 Build step 'Execute shell' marked build as failure
```

**Pull request merged**